### DISCUSSION TITLE
Install pdm in the full image.

### DISCUSSION THREAD

**Discussion by huxuan** [renovatebot/renovate#27401] [Suggest an Idea]
### Tell us more.

Currently, pdm is not installed in the full image. This leads to extra steps when trying to update lock file for pdm based projects. It would be better to have it preinstalled.

**Comment by rarkins** [comment]
@viceice @secustor sounds ok?

**Reply by secustor** [reply]
SGTM but we should write down gudelines when to not include tools in the full image and when not.

**Reply by rarkins** [reply]
I think we are mostly "all in" still

**Reply by viceice** [reply]
yes, at least the most used. we can add more, since the base image is now built separately. of cause new builds will need more time with each new tool

**Reply by huxuan** [reply]
I would appreciate to see this happen since the `pep621` manager already support pdm, but now it seems that the version is hard-coded [1].

[1] https://github.com/renovatebot/renovate/blob/e390cae1926b832ec79f7de1aa733caea710bbaa/lib/modules/manager/pep621/processors/pdm.spec.ts#L90

**Reply by rarkins** [reply]
You're pointing to a test file, it means nothing

**Reply by rarkins** [reply]
PRs welcome to add it to https://github.com/renovatebot/renovate/blob/main/tools/docker/Dockerfile

**Reply by huxuan** [reply]
Sorry I am not a typescript developer.

But maybe I think we should add it here [1]? Just behind the poetry.

[1] https://github.com/renovatebot/base-image/blob/706d3b6662c4988822a8bd49ec7c7d8ac8df06ce/Dockerfile#L62

**Reply by HonkingGoose** [reply]
pdm was added to the `renovatebot/base-image` with these PRs:

- https://github.com/renovatebot/base-image/pull/425
- https://github.com/renovatebot/base-image/pull/435 (backport to `v2`)
